pinot-core/src/main/java/org/apache/pinot/core/indexsegment/mutable/MutableSegmentImpl.java

```diff
@@ -108,19 +110,8 @@
   private final int _partitionId;
   private final boolean _nullHandlingEnabled;
 
-  // TODO: Keep one map to store all these info
-  private final Map<String, NumValuesInfo> _numValuesInfoMap = new HashMap<>();
-  private final Map<String, BaseMutableDictionary> _dictionaryMap = new HashMap<>();
-  private final Map<String, MutableForwardIndex> _forwardIndexMap = new HashMap<>();
-  private final Map<String, InvertedIndexReader> _invertedIndexMap = new HashMap<>();
-  private final Map<String, InvertedIndexReader> _rangeIndexMap = new HashMap<>();
-  private final Map<String, InvertedIndexReader> _textIndexMap = new HashMap<>();
-  private final Map<String, BloomFilterReader> _bloomFilterMap = new HashMap<>();
-  // Only store min/max for non-dictionary fields
-  private final Map<String, Comparable> _minValueMap = new HashMap<>();
-  private final Map<String, Comparable> _maxValueMap = new HashMap<>();
-
-  private final Map<String, RealtimeNullValueVectorReaderWriter> _nullValueVectorMap = new HashMap<>();
+  private final Map<String, IndexContainer> _indexContainerMap = new HashMap<>();
```

Comment:
should this be IndexKey instead of String?

Reply:
No, it is the column name. `IndexContainer` contains all the indexes and extra stats for a column.